pennylane/beta/queuing/base.py

```diff
@@ -0,0 +1,909 @@
+
+        For immutable nodes this method is called the first time :meth:`BaseQNode.evaluate`
+        or :meth:`.JacobianQNode.jacobian` is called, and for mutable nodes *each time*
+        they are called. It executes the quantum function,
+        stores the resulting sequence of :class:`.Operator` instances,
+        converts it into a circuit graph, and creates the Variable mapping.
+
+        .. note::
+           The Variables are only required for analytic differentiation,
+           for evaluation we could simply reconstruct the circuit each time.
+
+        Args:
+            args (tuple[Any]): Positional arguments passed to the quantum function.
+                During the construction we are not concerned with the numerical values, but with
+                the nesting structure.
+                Each positional argument is replaced with a :class:`~.Variable` instance.
+            kwargs (dict[str, Any]): Auxiliary arguments passed to the quantum function.
+        """
+        # TODO: Update the docstring to reflect the kwargs and the raising conditions
+        # pylint: disable=attribute-defined-outside-init, too-many-branches, too-many-statements
+
+        self.arg_vars, self.kwarg_vars = self._make_variables(args, kwargs)
+
+        ### Beta
+        """
+        # temporary queues for operations and observables
+        # TODO rename self.queue to self.op_queue
+        self.queue = []  #: list[Operation]: applied operations
+        self.obs_queue = []  #: list[Observable]: applied observables
+        """
+
+        ### Beta
+        # Creating the internal queue
+        self.queue = AnnotatedQueue()
+
+        # set up the context for Operator entry
+        with self:
+            try:
+                # generate the program queue by executing the quantum circuit function
+                if self.mutable:
+                    # it's ok to directly pass auxiliary arguments since the circuit is re-constructed each time
+                    # (positional args must be replaced because parameter-shift differentiation requires Variables)
+                    res = self.func(*self.arg_vars, **kwargs)
+                else:
+                    # TODO: Maybe we should only convert the kwarg_vars that were actually given
+                    res = self.func(*self.arg_vars, **self.kwarg_vars)
+            except:
+                # The qfunc call may have failed because the user supplied bad parameters, which is why we must wipe the created Variables.
+                self.arg_vars = None
+                self.kwarg_vars = None
+                raise
+
+
+        ### Beta
+        """
+        # check the validity of the circuit
+        self._check_circuit(res)
+
+        del self.queue
+        del self.obs_queue
+        """
+
+        ## Beta --- begin
+        # Removed checking when populating AQ
+        # Moved the important logic from _check_circuit
```

Comment:
This is related to #630: the creation of `self.ops` by merging the queue of operations and observables has now been moved to `_construct`.